ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/ValidatorExitsValidator.java

```diff
@@ -40,17 +49,78 @@ public boolean validateValidatorExitParameter(
         final Optional<List<ValidatorExit>> validatorExits) {
       return validatorExits.isEmpty();
     }
+
+    @Override
+    public boolean validateExitsInBlock(
+        final Block block, final List<ValidatorExit> validatorExits) {
+      final Optional<List<ValidatorExit>> maybeExits = block.getBody().getExits();
+      if (maybeExits.isPresent()) {
+        LOG.warn("Block {} contains exits but exits are prohibited", block.getHash());
+        return false;
+      }
+
+      if (block.getHeader().getExitsRoot().isPresent()) {
+        LOG.warn("Block {} header contains exits_root but exits are prohibited", block.getHash());
+        return false;
+      }
+
+      return true;
+    }
   }
 
   /** Used after Prague */
   class AllowedExits implements ValidatorExitsValidator {
 
+    private static final Logger LOG = LoggerFactory.getLogger(AllowedExits.class);
+
     @Override
     public boolean validateValidatorExitParameter(
         final Optional<List<ValidatorExit>> validatorExits) {
-      // TODO implement any extra required validation (see
-      // https://github.com/hyperledger/besu/issues/6800)
       return validatorExits.isPresent();
     }
+
+    @Override
+    public boolean validateExitsInBlock(
+        final Block block, final List<ValidatorExit> expectedExits) {
+      final Hash blockHash = block.getHash();
+
+      if (block.getHeader().getExitsRoot().isEmpty()) {
+        LOG.warn("Block {} must contain exits_root", blockHash);
+        return false;
+      }
+
+      if (block.getBody().getExits().isEmpty()) {
+        LOG.warn("Block {} must contain exits (even if empty list)", blockHash);
+        return false;
+      }
+
+      final List<ValidatorExit> exitsInBlock = block.getBody().getExits().get();
+      // TODO Do we need to allow for customization? (e.g. if the value changes in the next fork)
```

Comment:
Suggestion: We could rename it to PrageExitsValidator, and make it a standalone class. If the next fork needs to change this we can easily abstract a class override what we need.

Reply:
nit: Also, giving the validations, the semantics here are more like a `RequiredExitsValidator` than an `AllowedExistsValidator`.